### PR TITLE
[4.0] com_templates: Validate input for copy template form

### PR DESCRIPTION
**Pull request by YatharthVyas**
Pull Request for Issue #33433

### Summary of Changes
Added a missing required attribute to the input.

### Testing Instructions
1. Joomla Admin Panel -> System
2. Site Templates
3. Select a template
4. Click on the Copy Template button
5. Submit the form without entering anything in the input box

> Please enter Russian letters, for example, "ыыыыыыы" as the template name. The page will reload, there will be no error messages, but the style will not be copied either.

### Actual result BEFORE applying this Pull Request
Page reloads without showing any error.

### Expected result AFTER applying this Pull Request
image

For Russian Characters: 

> ыыыыыыы

image

### Documentation Changes Required
None

### CODE REVIEW

**Comment by Kostelano** [comment]
Please enter Russian letters, for example, "ыыыыыыы" as the template name. The page will reload, there will be no error messages, but the style will not be copied either.

Everything is in order with the Latin letters.

I think it makes sense to implement an error message, and not just validate the field for characters / letters.

**Comment by YatharthVyas** [comment]
@Kostelano Thanks, I missed out on this edge case.

Apparently, some non-English characters bypass the required check and they are passed as an empty string to the POST request. This causes the else block of strlen> 0 to execute. I've used this condition to display an error message. The current error message doesn't specify '**English** characters' so if there is a need to rephrase that, please let me know

russianname1

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 88c3c6a08b17fc5225fdc849ea019b5f3a605773<br /><br />Thanks, everything is working as it should. I translated this line into russian for Joomla 3, adding the phrase "only latin letters".<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33444">issues.joomla.org/tracker/joomla-cms/33444</a>.</sub>

File: administrator/components/com_templates/src/Controller/TemplateController.php [hunk 1]
```diff
@@ -240,6 +240,10 @@ public function copy()
 
 			return true;
 		}
+		else
+		{
+			$this->setMessage(Text::sprintf('COM_TEMPLATES_ERROR_INVALID_TEMPLATE_NAME'), 'error');
+		}
 
 		return false;
```
Comment (Quy) [inline administrator/components/com_templates/src/Controller/TemplateController.php]:
```suggestion
		}

		$this->setMessage(Text::sprintf('COM_TEMPLATES_ERROR_INVALID_TEMPLATE_NAME'), 'error');

		return false;
```

**Comment by brianteeman** [comment]
>  translated this line into russian for Joomla 3, adding the phrase "only latin letters".

Yes I think thats the best way to address this.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 8d6d1d74faecb690674c37bbb721054e53dbddf0<br /><br />@Kostelano can you please retest? Issue tracker doesn't show you test anymore.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33444">issues.joomla.org/tracker/joomla-cms/33444</a>.</sub>

**Comment by alikon** [comment]
that's normal because a new commit has been made after that test

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 8d6d1d74faecb690674c37bbb721054e53dbddf0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33444">issues.joomla.org/tracker/joomla-cms/33444</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33444">issues.joomla.org/tracker/joomla-cms/33444</a>.</sub>

**Comment by ghost** [comment]
> that's normal because a new commit has been made after that test

Thanks @alikon. I asked cause if i test i don't look later if it don't count cause of new commit so i'm fine if someone inform me about retesting.

**Comment by richard67** [comment]
Restarted drone. All fine now.

**Comment by Quy** [comment]
Thank you!

**Pull request merged**